lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_action.rb

```diff
@@ -15,16 +16,71 @@ class FirebaseAppDistributionAction < Action
       FIREBASECMD_ACTION = "appdistribution:distribute".freeze
       BASE_URL = "https://firebaseappdistribution.googleapis.com"
       PATH = "/v1alpha/apps/"
+      MAX_POLLING_RETRIES = 60
+      POLLING_INTERVAL_S = 2
 
       extend Helper::FirebaseAppDistributionHelper
 
       def self.run(params)
         params.values # to validate all inputs before looking for the ipa/apk
-        validate_app!(params[:app])
+        app_id = nil
+        platform = Actions.lane_context[Actions::SharedValues::PLATFORM_NAME]
+        binary_path = Shellwords.escape(params[:ipa_path] || params[:apk_path])
+
+        if params[:app] # Set app_id if it is specified as a parameter
+          app_id = params[:app]
+        else
+          if platform == :ios
+            archive_path = Actions.lane_context[SharedValues::XCODEBUILD_ARCHIVE]
+            if archive_patharchive_path
+              app_id = findout_ios_app_id_from_archive(archive_path)
+            end
+          end
+          if app_id.nil?
+            app_id = UI.input("Your app's Firebase App ID. You can find the App ID in the Firebase console, on the General Settings page: ")
+          end
+        end
+
+        upload_token = get_upload_token(app_id, binary_path)
```

Comment:
Before we polling we should make an initial `upload_status` call outside of the loop. If that returns `SUCCESS` then we should display "This APK has been uploaded before. Skipping upload step." This is what we do in the gradle plugin (see attached screenshot).
<img width="693" alt="Screen Shot 2020-06-22 at 3 32 13 PM" src="https://user-images.githubusercontent.com/401051/85327801-92900a80-b49d-11ea-8c34-d11b2d66e177.png">